packages/core/docs/general/cms.md

```diff
@@ -0,0 +1,139 @@
+
+```typescript
+// public API
+['@vue-storefront/{cms}/nuxt', config]
+```
+
+### Content fetching
+
+Another (and the most important) thing that we need is a composable to fetch the content from our CMS. To create this composable you should use `useContentFactory` from Vue Storefront core.
+
+```typescript
+// integration
+import { useContentFactory } from '@vue-storefront/core'
+
+// CONTENT and CONTENT_SEARCH_PARAMS are your CMS-specific types, 
+// we advise to have at least 'id' param for search
+const useContent = useContentFactory<CONTENT, CONTENT_SEARCH_PARAMS>({
+  // (params: CONTENT_SEARCH_PARAMS) => Promise<CONTENT>;
+  search (params) { 
+    // write your content fetching logic here
+    return content
+  }
+}) 
+```
+
+The public API:
+
+```typescript
+const { search, content, loading, error } = useContent()
+```
+
+### Content rendering
+
+Many CMS systems allow to control the page layout by returning a list of components as a JSON file. 
+These components are then rendered in the application. 
+The component that is mapping the content object into Vue components and renders them is called `RenderContent`.
+It's created from `renderContentFactory` from Vue Storefront core.
+
+So, what `renderContentFactory` does is rendering your content data into the Vue components.
+To be precise - your data will be converted to the components *Array*. 
+
+This factory requires special structure, and we have to provide it while using it.
+
+```typescript
+interface RenderComponent {
+  componentName: string;
+  props?: {};
+}
+```
+
+To pass this data we can create special composable and use `extractContent` parameter.
+
+```typescript
+// integration
+import { renderComponentFactory } from '@vue-storefront/core'
+
+// CONTENT and CONTENT_SEARCH_PARAMS are your CMS-specific types, 
+// we advise to have at least 'id' param for search
+const RenderComponent = renderComponentFactory({
+  // (extractContent: CONTENT) => { componentName, props }[]
+  extractContent (content) { 
+    // extract an array of { componentName: string, props: { name, value } } 
+    // pairs and the factory will generate a component that will render these 
+    // components based on the ones that are currently registered within application
+    return components[]
+  }
+}) 
+```
+
+And the real life example.
+
+```typescript
+<template>
+  <div v-if="loading">Loading content...</div>
+  <div v-if="error">Something is wrong!</div>
+  <RenderContent v-if="content" :content="content" />
+</template>
+
+<script>
+import { onSSR } from '@vue-storefront/core'
+import { useContent, RenderContent } from '@vue-storefront/my-super-cms'
+// These are the components that will be rendered by RenderContent.
+// `extractContent` should return [{ componentName: 'CMSBanner', props: { ... }}, { componentName: 'CMSHero', props: { ... }}]
+import { CMSBanner, CMSHero } from '~/components'
+
+export default {
+  components: {
+    RenderContent,
+    CMSBanner,
```

Comment:
Could we lazy load them? Let's imagine `CMSHero` will be rendered/or not based on some setting in CMS. If we could lazy load it then it would not be loaded if not necessary. I think it would be nice to put lazy-load in an example or info about it if it is possible

Reply:
The implementation is more complex, for sure you will need some kind of handlers/wrapper to do that. Then it can be handled along with it.